Artifacts/windows-ConfigurationManagerCurrentBranch(EVAL)/Reset-SqlAdmin.psm1

```diff
@@ -0,0 +1,338 @@
+		try { $checkcluster = Get-Service -ComputerName $ipaddr | Where-Object { $_.Name -eq "ClusSvc" -and $_.Status -eq "Running" } }
+		catch { throw "Can't check services. Check permissions and firewall." }
+				
+		if ($checkcluster -ne $null) {
+			$clusterResource = Get-WmiObject -Authentication PacketPrivacy -Impersonation Impersonate -class "MSCluster_Resource" -namespace "root\mscluster" -computername $hostname | 
+			Where-Object { $_.Name.StartsWith("SQL Server") -and $_.OwnerGroup -eq "SQL Server ($instance)" }
+		}
+		
+		# Take SQL Server offline so that it can be started in single-user mode
+		if ($clusterResource.count -gt 0) {
+			$isclustered = $true
+			try { $clusterResource | Where-Object { $_.Name -eq "SQL Server" } | ForEach-Object { $_.TakeOffline(60) } }
+			catch {
+				$clusterResource |  Where-Object { $_.Name -eq "SQL Server" }  | ForEach-Object { $_.BringOnline(60) }
+				$clusterResource |  Where-Object { $_.Name -ne "SQL Server" }  | ForEach-Object { $_.BringOnline(60) }
+				throw  "Could not stop the SQL Service. Restarted SQL Service and quit."
+			}
+		} else {
+			try { 
+				Stop-Service -InputObject $sqlservice -Force 
+				Write-Output "Successfully stopped SQL service"
+			} catch {
+				Start-Service -InputObject $instanceservices
+				throw  "Could not stop the SQL Service. Restarted SQL service and quit."
+			}
+		}
+
+		# /mReset-SqlAdmin Starts an instance of SQL Server in single-user mode and only allows this script to connect.
+		Write-Output "Starting SQL Service from command line"
+		try {
+			if ($hostname -eq $env:COMPUTERNAME) { 
+				$netstart = net start ""$displayname"" /mReset-SqlAdmin 2>&1
+				if ("$netstart" -notmatch "success") { throw }
+			} 
+			else {
+				$netstart = Invoke-Command -ErrorAction Stop -Session $session -Args $displayname -ScriptBlock { net start ""$args"" /mReset-SqlAdmin } 2>&1
+				foreach ($line in $netstart) { 
+					if ($line.length -gt 0) { Write-Output $line }
+				}
+			}
+		} catch {
+			Stop-Service -InputObject $sqlservice -Force -ErrorAction SilentlyContinue
+			
+			if ($isclustered) {
+				$clusterResource | Where-Object Name -eq "SQL Server" | ForEach-Object { $_.BringOnline(60) }
+				$clusterResource | Where-Object Name -ne "SQL Server" | ForEach-Object { $_.BringOnline(60) }
+			} else {
+				Start-Service -InputObject $instanceservices -ErrorAction SilentlyContinue
+			}
+			throw "Couldn't execute net start command. Restarted services and quit."
+		}
+
+		Write-Output "Reconnecting to SQL instance"
+		try { Invoke-ResetSqlCmd -SqlServer $sqlserver -Sql  "SELECT 1"  | Out-Null } 
+		catch {
+			try {
+				Start-Sleep 3
+				Invoke-ResetSqlCmd -SqlServer $sqlserver -Sql  "SELECT 1" | Out-Null
+			} catch {
+				Stop-Service Input-Object $sqlservice -Force -ErrorAction SilentlyContinue
+				if ($isclustered) { 
+					$clusterResource |  Where-Object { $_.Name -eq "SQL Server" } | ForEach-Object { $_.BringOnline(60) }
```

Comment:
Same command twice in a row here too. If there is a reason, please just code comment it.